core/src/main/java/eu/drus/jpa/unit/decorator/jpa/PersistenceUnitDecorator.java

```diff
@@ -17,17 +17,17 @@ public int getPriority() {
     }
 
     @Override
-    public void beforeTest(final TestMethodInvocation invocation) throws Exception {
+    public void beforeTest(final TestInvocation invocation) throws Exception {
         final EntityManagerFactory emf = (EntityManagerFactory) invocation.getContext().getData(Constants.KEY_ENTITY_MANAGER_FACTORY);
 
         final Class<?> fieldType = invocation.getContext().getPersistenceField().getType();
         if (fieldType.equals(EntityManagerFactory.class)) {
-            injectValue(invocation.getContext().getPersistenceField(), invocation.getTestInstance(), emf);
+            injectValue(invocation.getContext().getPersistenceField(), invocation.getTestInstance().get(), emf);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)

Reply:
if the required is not present an exception has to be thrown anyway.